assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -90,7 +102,118 @@ describe("DiversionPage", () => {
     expect(screen.queryByTitle("Detour End")).not.toBeInTheDocument()
   })
 
-  test("directions start with origin intersection when second waypoint is added", async () => {
+  test("has a normal cursor before placing the start point", async () => {
+    const { container } = render(<DiversionPage />)
+
+    await waitFor(() => {
+      expect(
+        container.querySelector(".c-detour_map--map__clickable")
+      ).not.toBeInTheDocument()
+    })
+  })
+
+  test("has a waypoint cursor, given by the __clickable class, once the start point is placed", async () => {
+    const { container } = render(<DiversionPage />)
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    await waitFor(() => {
+      expect(
+        container.querySelector(".c-detour_map--map__clickable")
+      ).toBeInTheDocument()
+    })
+  })
+
+  test("has no waypoints when placing the start point", async () => {
+    const { container } = render(<DiversionPage />)
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    await waitFor(() => {
+      expect(screen.getByTitle("Detour Start")).toBeVisible()
+    })
+
+    expect(
+      container.querySelectorAll(".c-detour_map-circle-marker--detour-point")
+    ).toHaveLength(0)
+  })
+
+  test("can click on map to add a waypoint", async () => {
+    const { container } = render(<DiversionPage />)
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+    act(() => {
+      fireEvent.click(container.querySelector(".c-vehicle-map")!)
+    })
+
+    await waitFor(() => {
+      expect(
+        container.querySelectorAll(".c-detour_map-circle-marker--detour-point")
+      ).toHaveLength(1)
+    })
+  })
+
+  test("directions starts out with placeholder text", async () => {
+    render(<DiversionPage />)
+
+    const { getByText } = within(
+      screen
+        .getByRole("heading", { name: "Detour Directions" })
+        .closest("section")!
+    )
+
+    await waitFor(() => {
+      expect(
+        getByText(
+          "Click a point on the regular route to start drawing your detour. As you continue to select points on the map, turn-by-turn directions will appear in this panel."
+        )
+      ).toBeVisible()
+    })
+  })
+
+  test("calls the API when the first waypoint is added", async () => {
+    const { container } = render(<DiversionPage />)
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
```

Comment:
question(non-blocking): why `not.toBeNull` here?

Reply:
I think Github got confused again :(